### PR TITLE
FE-884 - Component Wrapper for Matchbox ProgressBar

### PR DESCRIPTION
**Pull request by shrirupa**
FE-884 - Component Wrapper for Matchbox ProgressBar

### What Changed
 - ProgressBar has a single, globally accessible point of entry within the app
 - ProgressBar is only imported from Matchbox in one place - replaced all instances of the ProgressBar directly imported from matchbox with this new global component
- New props available to the Hibana version of the component are exposed - and subsequently ignored by the original component (https://jira.int.messagesystems.com/browse/FE-812)
- Ensure no unusual positioning UI defects are introduced as a result of this change - both for when using Hibana and when using the default theme

### How To Test
 - enable hibana ui flag 
`{
	"options": {
		"ui":{
"hibana":{
	"hasThemeControls": true
}
		}
	}
 }`
- Check all instances of ProgressBar for ui defects with and without hibana enabled

### To Do
- [ ] Address feedback

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
     "date-fns": "^1.28.5",
     "downshift": "^3.2.10",
     "email-addresses": "^3.0.1",
+    "hibana": "npm:@sparkpost/matchbox@^4.0.0-hibana.3",
```
Comment (kchu1337) [inline package.json]: Hibana is already imported in line 72

File: src/components/matchboxWrapper/ProgressBar.container.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { ProgressBar as PreHibanaProgressBar } from '@sparkpost/matchbox';
+import { ProgressBar as HibanaProgressBar } from 'hibana';
+import { useHibana } from 'src/context/HibanaContext';
+import _ from 'lodash';
+
+export default function ProgressBar(props) {
+  const [state, _dispatch] = useHibana();
```
Comment (kchu1337) [inline src/components/matchboxWrapper/ProgressBar.container.js]: You don't actually need to deconstruct the 2nd argument.

File: src/components/matchboxWrapper/tests/ProgressBar.container.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        .last()
+        .props(),
+    ).not.toHaveProperty('size');
+    expect(
+      instance
+        .find('ProgressBar')
+        .last()
+        .props(),
+    ).not.toHaveProperty('my');
+  });
```
Comment (kchu1337) [inline src/components/matchboxWrapper/tests/ProgressBar.container.test.js]: Could also add a shallow test to check that hibana progressbar is returned vs old progressbar

File: src/components/matchboxWrapper/tests/ProgressBar.container.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+describe('ProgressBar', () => {
+  const subject = (props, context) => {
+    return mount(
+      <HibanaProvider>
+        <ProgressBar {...props} />
+      </HibanaProvider>,
+      context,
+    );
+  };
+  it('should only pass pre-hibana when hibana is not enabled', () => {
```
Comment (kchu1337) [inline src/components/matchboxWrapper/tests/ProgressBar.container.test.js]: This test is checking that the progress bar only contains old props right? Might want to rename the test to something like `should only pass pre-hibana props when hibana is not enabled`

File: src/components/matchboxWrapper/ProgressBar.container.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (nicklemmon) [inline src/components/matchboxWrapper/ProgressBar.container.js]: I don't think we want to use the `.container` suffix - that implies this is related to Redux? We are having that same convo over with Kevin's PR.

What do you think?

Comment (shrirupa) [reply src/components/matchboxWrapper/ProgressBar.container.js]: sure, i just renamed it 👍 .

File: package-lock.json [hunk 6]
```diff
@@ -3888,13 +3888,14 @@
       }
     },
     "@sparkpost/matchbox-hibana": {
-      "version": "npm:@sparkpost/matchbox@4.0.0-hibana.3",
-      "resolved": "https://registry.npmjs.org/@sparkpost/matchbox/-/matchbox-4.0.0-hibana.3.tgz",
-      "integrity": "sha512-TuXvCJxOkN4VKPTxO1Bzm0oAnuPnT9nYcZ48ehY0KJJarz+PxtzRzQRjfQYrV7aO89hIYhPDeo90Pw0noTAAsg==",
+      "version": "npm:@sparkpost/matchbox@4.0.0-hibana.4",
```
Comment (kchu1337) [inline package-lock.json]: Could revert these diffs. I assume these were from when you installed hibana?

File: src/components/matchbox/ProgressBar.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { ProgressBar as PreHibanaProgressBar } from '@sparkpost/matchbox';
```
Comment (kchu1337) [inline src/components/matchbox/ProgressBar.js]: Could rename to `OGProgressBar` but meh.

File: package-lock.json (hunk 6 above)
Comment (shrirupa) [reply package-lock.json]: let's leave this one, since version of hibana seems to have updated from 3 to 4

**Pull request merged**